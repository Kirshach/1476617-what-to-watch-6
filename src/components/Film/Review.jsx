import React from 'react';
import PropTypes from 'prop-types';

import {formatDate, formatCommentRating} from './_helpers';

const Review = ({
  comment,
  rating,
  date,
  username
}) => {
  const parsedDate = new Date(date);
  const formattedDate = formatDate(parsedDate);
  const formattedRating = formatCommentRating(rating);
  return (
    <div className="review">
      <blockquote className="review__quote">
        <p className="review__text">{comment}</p>

        <footer className="review__details">
          <cite className="review__author">{username}</cite>
          <time className="review__date" dateTime={parsedDate.toISOString()}>{formattedDate}</time>
        </footer>
      </blockquote>

      <div className="review__rating">{formattedRating}</div>
    </div>
  );
};

Review.propTypes = {
  comment: PropTypes.string.isRequired,
  rating: PropTypes.number.isRequired,
  date: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
};

export default Review;
