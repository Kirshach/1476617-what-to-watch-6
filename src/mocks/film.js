export const macbeth = {
  name: `Macbeth`,
  posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Macbeth.jpg`,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/macbeth.jpg`,
  backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Macbeth.jpg`,
  backgroundColor: `#F1E9CE`,
  description: `Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.`,
  rating: 3.3,
  scoresCount: 48798,
  director: `Justin Kurzel`,
  starring: [
    `Michael Fassbender`,
    `Marion Cotillard`,
    `Jack Madigan`
  ],
  runTime: 113,
  genre: `Drama`,
  released: 2015,
  id: 1,
  isFavorite: true,
  videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
};

export const bronson = {
  name: `Bronson`,
  posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
  backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
  backgroundColor: `#73B39A`,
  description: `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
  rating: 3.6,
  scoresCount: 109661,
  director: `Nicolas Winding Refn`,
  starring: [
    `Tom Hardy`,
    `Kelly Adams`,
    `Luing Andrews`
  ],
  runTime: 92,
  genre: `Action`,
  released: 2008,
  id: 9,
  isFavorite: false,
  videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
};

export const revenant = {
  name: `The Revenant`,
  posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Revenant.jpg`,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/revenant.jpg`,
  backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Revenant.jpg`,
  backgroundColor: `#92918B`,
  description: `A frontiersman on a fur trading expedition in the 1820s fights for survival after being mauled by a bear and left for dead by members of his own hunting team.`,
  rating: 4,
  scoresCount: 618498,
  director: `Alejandro G. Iñárritu`,
  starring: [
    `Leonardo DiCaprio`,
    `Tom Hardy`,
    `Will Poulter`
  ],
  runTime: 156,
  genre: `Action`,
  released: 2015,
  id: 8,
  isFavorite: false,
  videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
};
