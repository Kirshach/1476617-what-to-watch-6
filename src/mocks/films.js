export const films = [{
  id: 1,
  name: `The Green Elephant`,
  posterImage: `https://upload.wikimedia.org/wikipedia/ru/3/34/%D0%97%D0%B5%D0%BB%D1%91%D0%BD%D1%8B%D0%B9_%D1%81%D0%BB%D0%BE%D0%BD%D0%B8%D0%BA.jpg`,
  previewImage: `http://avatars.mds.yandex.net/get-kino-vod-films-gallery/177294/8edaf90de249e40a839c77ced441aab8/280x178_2`,
  backgroundImage: `https://cdnimg.rg.ru/img/content/155/11/95/1_d_850.jpg`,
  backgroundColor: `#ffffff`,
  videoLink: `https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ForBiggerEscapes.mp4`,
  previewVideoLink: `https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ForBiggerBlazes.mp4`,
  description: `Two junior officers in a guardhouse cell are having to solve daunting challenges and psychological questions in a limited space`,
  rating: 10.0,
  scoresCount: 228,
  director: `Svetlana Baskova`,
  starring: [`Vladimir Epiphatsev`, `Sergey Pakhomov`, `Aleksander Maslaev`, `Anatoliy Osmolovskii`],
  runTime: 99,
  genre: `Comedies`,
  released: 1999,
  isFavorite: true
}, {
  id: 2,
  name: `Vase de noces`,
  posterImage: `https://www.ecranlarge.com/uploads/image/001/142/8ryja656mssrbsuw7hjlz830rtz-220.jpg`,
  previewImage: `https://sun9-22.userapi.com/c623429/v623429104/33609/wveCFHFXRzg.jpg`,
  backgroundImage: `https://ic.pics.livejournal.com/pasha_popolam/77543864/375839/375839_900.jpg`,
  backgroundColor: `#ffffff`,
  videoLink: `https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/Sintel.mp4`,
  previewVideoLink: `https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4`,
  description: `An art-house black and white film without dialogue. It's about a man who lives alone on a farm with his pigs. He falls in love with his sow, has offscreen sex with her, and later she gives birth to human/porcine hybrids`,
  rating: 9.9,
  scoresCount: 1488,
  director: `Thierry Zéno`,
  starring: [`Dominique Garny`, `Sexy ass piggie`, `Some geese`],
  runTime: 80,
  genre: `Dramas`,
  released: 1974,
  isFavorite: true
}, {
  id: 3,
  name: `The Human Centipede`,
  posterImage: `https://images-na.ssl-images-amazon.com/images/I/91Zd7OiKbML._SL1500_.jpg`,
  previewImage: `https://i.guim.co.uk/img/static/sys-images/Guardian/About/General/2011/9/13/1315921510309/Still-from-the-Human-Cent-007.jpg?width=300&quality=45&auto=format&fit=max&dpr=2&s=74aee547f36d9b44b278e9a5dcbf2f97`,
  backgroundImage: `http://2.bp.blogspot.com/-j8JdzITKVXA/UnSN6Q7x02I/AAAAAAAAAiY/pdVJrqQUCl8/s1600/Human+centipede+feed.jpg`,
  backgroundColor: `#ffffff`,
  videoLink: `https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ElephantsDream.mp4`,
  previewVideoLink: `https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ForBiggerFun.mp4`,
  description: `A mad scientist kidnaps and mutilates a trio of tourists in order to reassemble them into a human centipede, created by stitching their mouths to each others' rectums.`,
  rating: 9.8,
  scoresCount: 420,
  director: `Tom Six`,
  starring: [`Dieter Laser`, `Ashley C. Williams`, `Ashlynn Yennie`],
  runTime: 110,
  genre: `Documentary`,
  released: 2009,
  isFavorite: true
}, {
  id: 4,
  name: `Gayniggers from Outer Space`,
  posterImage: `https://avatars.mds.yandex.net/get-kinopoisk-image/1773646/882998f1-9a98-4fd3-ac0e-dedc0b7fd160/300x450`,
  previewImage: `https://resizer.mail.ru/p/1d3b1a49-8a96-5d95-ae37-fa871d85565b/AAACY90FK9qRp4B4z90Nlw3g-fsPNMokZxvfQ5iVZcu9kp6fSm2lQjCfYlt1F4-sCY1Tud9-MfJz3AJ8OsXfZQi9ULc.jpg`,
  backgroundImage: `https://m.media-amazon.com/images/M/MV5BZGQ4NjZhNjAtZGViZi00NTNkLTkxZWUtNDU2YjYyNjc4OGM2XkEyXkFqcGdeQXVyNzI2NzgzMzc@._V1_.jpg`,
  backgroundColor: `#ffffff`,
  videoLink: `https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ForBiggerJoyrides.mp4`,
  previewVideoLink: `https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/Sintel.mp4`,
  description: `Extraterrestrial beings travel the galaxy to free men "oppressed" by females to make way for an entirely-homosexual society.`,
  rating: 9.7,
  scoresCount: 69,
  director: `Morten Lindberg`,
  starring: [`Coco C.P. Dalbert`, `Sammy Salomon`, `Gbatokai Dakinah`],
  runTime: 26,
  genre: `Crime`,
  released: 1992,
  isFavorite: false
}, {
  id: 5,
  name: `The Human Centipede 2`,
  posterImage: `https://images-na.ssl-images-amazon.com/images/I/81BhPvzTN%2BL._SL1500_.jpg`,
  previewImage: `https://i.guim.co.uk/img/static/sys-images/Guardian/About/General/2011/9/13/1315921323418/Still-from-the-Human-Cent-007.jpg?width=300&quality=45&auto=format&fit=max&dpr=2&s=3ce4dbb98715a319444f89dc2d03abad`,
  backgroundImage: `https://static01.nyt.com/images/2011/10/02/arts/02JpCentipede1_span/02JpCentipede1-articleLarge.jpg?quality=75&auto=webp&disable=upscale`,
  backgroundColor: `#ffffff`,
  videoLink: `https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ForBiggerJoyrides.mp4`,
  previewVideoLink: `https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/Sintel.mp4`,
  description: `Inspired by the fictional Dr. Heiter, disturbed loner Martin dreams of creating a 12-person centipede and sets out to realize his sick fantasy.`,
  rating: 9.9,
  scoresCount: 9000,
  director: `Tom Six`,
  starring: [`Laurence R. Harvey`, `Ashlynn Yennie`, `Maddi Black`],
  runTime: 92,
  genre: `Kids & Family`,
  released: 2011,
  isFavorite: true
}, {
  id: 6,
  name: `A Clockwork Orange`,
  posterImage: `https://cdn.britannica.com/61/197661-050-A3F3C598/Movie-poster-A-Clockwork-Orange-Stanley-Kubrick.jpg`,
  previewImage: `https://www.broadway.org.uk/sites/default/files/styles/banner_crop/public/2019-03/a-clockwork-orange-banner-min.jpg?h=7683c7ce&itok=Bjkc949f`,
  backgroundImage: `http://screen-slate-rails.herokuapp.com/system/images/W1siZiIsIjIwMTgvMTIvMjEvMTgvMzMvMTQvMmRmNjI1ZDgtODZhNi00YjFiLThiNDItNDgyZGZlNGQ0MTM2L2EtY2xvY2t3b3JrLW9yYW5nZS5qcGciXV0/a-clockwork-orange.jpg?sha=f8257dd54f25465d`,
  backgroundColor: `#ffffff`,
  videoLink: `https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/VolkswagenGTIReview.mp4`,
  previewVideoLink: `https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/WhatCarCanYouGetForAGrand.mp4`,
  description: `In the future, a sadistic gang leader is imprisoned and volunteers for a conduct-aversion experiment, but it doesn't go as planned.`,
  rating: 9.8,
  scoresCount: 13,
  director: `Stanley Kubrick`,
  starring: [`Malcolm McDowell`, `Patrick Magee`, `Michael Bates`],
  runTime: 102,
  genre: `Kids & Family`,
  released: 1971,
  isFavorite: false
}, {
  id: 7,
  name: `Man Bites Dog`,
  posterImage: `https://s3.amazonaws.com/criterion-production/films/6727067d1f4e46a650902bc011e19592/tIr6988MSCKOQmR6KedSRIzbetNOjp_large.jpg`,
  previewImage: `https://imgtoolkit.culturebase.org/?color=FFFFFF&quality=8&ar_ratio=1.3&format=jpg&file=http%3A%2F%2Fdata.heimat.de%2Fpics%2Fe%2F8%2Ff%2F1%2F5%2Fpic_1506694613_e8f152e870811472cd24581b2f22e7f5.jpeg&do=cropOut&width=1200&height=780`,
  backgroundImage: `https://i.kinja-img.com/gawker-media/image/upload/s--fCsYZ5nd--/c_scale,fl_progressive,q_80,w_800/gblxqkhxrn0bjqihrgaf.jpg`,
  backgroundColor: `#ffffff`,
  videoLink: `https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/VolkswagenGTIReview.mp4`,
  previewVideoLink: `https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/WhatCarCanYouGetForAGrand.mp4`,
  description: `A film crew follows a ruthless thief and heartless killer as he goes about his daily routine. But complications set in when the film crew lose their objectivity and begin lending a hand.`,
  rating: 7.2,
  scoresCount: 1389798721938789794,
  director: `Rémy Belvaux`,
  starring: [`Benoît Poelvoorde`, `Jacqueline Poelvoorde-Pappaert`, `Nelly Pappaert`],
  runTime: 111,
  genre: `Documentary`,
  released: 1992,
  isFavorite: false
}, {
  id: 8,
  name: `Begotten`,
  posterImage: `https://upload.wikimedia.org/wikipedia/en/thumb/a/a0/Begotten%2C_World_Artists_VHS_Cover%2C_March_1990.jpg/220px-Begotten%2C_World_Artists_VHS_Cover%2C_March_1990.jpg`,
  previewImage: `https://i0.wp.com/moviesandmania.com/wp-content/uploads/2013/01/begotten-3.png?resize=499%2C374&ssl=1`,
  backgroundImage: `https://i0.wp.com/www.splittoothmedia.com/wp-content/uploads/2019/10/Begotten.png?fit=631%2C475&ssl=1`,
  backgroundColor: `#ffffff`,
  videoLink: `https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/VolkswagenGTIReview.mp4`,
  previewVideoLink: `https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/WhatCarCanYouGetForAGrand.mp4`,
  description: `Presented in a surreal, gory and entirely visual manner, Begotten tells of the death of religion, the abuse of nature by Man and a nihilistic outlook on what life ultimately is.`,
  rating: 100.0,
  scoresCount: 12381267361784617826412786381726378162397816287,
  director: `E. Elias Merhige`,
  starring: [`Brian Salzberg`, `Donna Dempsey`, `Stephen Charles Barry`],
  runTime: 72,
  genre: `Romance`,
  released: 1989,
  isFavorite: true
}];


