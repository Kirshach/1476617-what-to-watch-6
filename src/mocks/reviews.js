export const reviewData = {
  id: 1,
  user: {
    id: 13,
    name: `Zak`
  },
  rating: 1.4,
  comment: `This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.`,
  date: `2021-03-07T08:04:28.658Z`
};

const reviewsData = [
  reviewData,
  {
    id: 2,
    user: {
      id: 17,
      name: `Emely`
    },
    rating: 7.2,
    comment: `This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.`,
    date: `2021-02-22T08:04:28.658Z`
  },
];

export const reviews = {
  id: 1,
  data: reviewsData,
};
