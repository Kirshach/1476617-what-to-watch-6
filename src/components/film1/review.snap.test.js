import React from 'react';
import {render} from '@testing-library/react';

import {reviewData as review} from '../../mocks/reviews';
import Review from './review';

test(`<Review /> renders correctly`, () => {
  const {container} = render(
      <Review
        username={review.user.name}
        rating={review.rating}
        date={review.date}
        comment={ review.comment }
      />
  );
  expect(container).toMatchSnapshot();
});
