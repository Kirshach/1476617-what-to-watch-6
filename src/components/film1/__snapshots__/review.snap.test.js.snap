// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Review /> renders correctly 1`] = `
<div>
  <div
    class="review"
  >
    <blockquote
      class="review__quote"
    >
      <p
        class="review__text"
      >
        This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.
      </p>
      <footer
        class="review__details"
      >
        <cite
          class="review__author"
        >
          Zak
        </cite>
        <time
          class="review__date"
          datetime="2021-03-07T08:04:28.658Z"
        >
          March 7, 2021
        </time>
      </footer>
    </blockquote>
    <div
      class="review__rating"
    >
      1,4
    </div>
  </div>
</div>
`;
